cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -362,9 +363,10 @@ findProjectRoot = do
 readProjectConfig :: Verbosity -> FilePath -> Rebuild ProjectConfig
 readProjectConfig verbosity projectRootDir = do
     global <- readGlobalConfig verbosity
-    local  <- readProjectLocalConfig      verbosity projectRootDir
-    extra  <- readProjectLocalExtraConfig verbosity projectRootDir
-    return (global <> local <> extra)
+    local  <- readProjectLocalConfig       verbosity projectRootDir
+    freeze <- readProjectLocalFreezeConfig verbosity projectRootDir
```

Comment:
IIRC @tibbe was quite vehemently against having a separate `.freeze` file, that's why we only have `cabal.config` and not `cabal.freeze.config` in the standard path. See the discussion [here](https://github.com/haskell/cabal/pull/1519).